File: src/main/java/com/github/miyashy/karatesample/infrastructure/UserRepositoryImpl.java
package com.github.miyashy.karatesample.infrastructure;

import com.github.miyashy.karatesample.domain.User;
import com.github.miyashy.karatesample.domain.repository.UserRepository;
import org.springframework.stereotype.Repository;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

@Repository
public class UserRepositoryImpl implements UserRepository {
    private final Map<UUID, User> map = new HashMap<>();

    @Override
    public User save(User user) {
        map.put(user.getId(), user);
        return user;
    }

    @Override
    public User find(UUID id) {
        return map.get(id);
    }

    @Override
    public void delete(User user) {
        map.remove(user.getId());
    }
}

File: src/main/kotlin/com/github/miyashy/karatesample/infrastructure/UserRepositoryImpl.kt
package com.github.miyashy.karatesample.infrastructure

import com.github.miyashy.karatesample.domain.User
import com.github.miyashy.karatesample.domain.repository.UserRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
class UserRepositoryImpl: UserRepository{
    val map: MutableMap<UUID, User> = mutableMapOf()
    override fun save(user: User): User {
        map[user.id] = user
        return user
    }

    override fun find(id: UUID): User? {
        return map[id]
    }

    override fun delete(user: User) {
        map.remove(user.id)
    }
}